genie-web/src/main/java/com/netflix/genie/web/tasks/job/JobCompletionService.java

```diff
@@ -154,6 +156,19 @@ void handleJobCompletion(final JobFinishedEvent event) throws GenieException {
         final Map<String, String> tags = MetricsUtils.newSuccessTagsMap();
 
         try {
+            /*
+             * Get the cluster and command chosen for the job for tagging metrics.
+             * Swallowing any exception in getting those as this step is not
+             * critical and should not show that the entire job failed
+             */
+            try {
+                final Cluster cluster = jobSearchService.getJobCluster(jobId);
+                final Command command = jobSearchService.getJobCommand(jobId);
+                MetricsUtils.addCommonJobWorkflowMetricTags(cluster, command, tags);
+            } catch (final Exception e) {
```

Comment:
I think it's cleaner if you just handle the error and always put a tag. This way you can at least put a tag like "NO CLUSTER FOUND" or pass null to your utility method representing a cluster doesn't exist. This way you could so something like 
```
final Cluster cluster;
final Command command;

try {
    cluster = this.jobSearchService.getJobCluster(jobId);
} catch (final Exeption e) {
    log.warn("Unable to tag metrics with cluster due to {}, e.getMessage(), e);
    cluster = null;
}

try {
    command = this.jobSearchService.getJobCommand(jobId);
} catch (final Exeption e) {
    log.warn("Unable to tag metrics with command due to {}, e.getMessage(), e);
    command = null;
}

MetricsUtils.addCommonJobWorkflowMetricTags(cluster, command, tags);
```